runtime/usr/lib/python2.7/dist-packages/cloudinit/handlers/zalando_ami_config.py

```diff
@@ -37,4 +41,10 @@ def handle_part(self, _data, ctype, filename, payload, frequency):
 
             LOG.debug("Storing merged configuration...")
             config_yaml = util.yaml_dumps(config_merged)
-            util.write_file(TAUPAGE_CONFIG, config_yaml, 0o444)
+            util.write_file(TMP_TAUPAGE_CONFIG, config_yaml, 0o444)
+
+            LOG.debug("Comparing current configuration with the old one...")
+            subprocess.call(['diff', '-u0', TAUPAGE_CONFIG, TMP_TAUPAGE_CONFIG])
```

Comment:
Actually, should it not be throwing an error earlier when it tries to read the config with:
```
            LOG.debug("Loading existing configuration...")
            config_yaml = util.read_file_or_url(TAUPAGE_CONFIG)
```
?